cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -281,7 +295,9 @@ private PullAppResponse<?> pullAndValidateApplication(ApplicationReference appRe
    */
   public RepositoryAppsResponse listApps(NamespaceId namespace) throws NotFoundException,
     AuthenticationConfigException {
-    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.GET);
+    accessEnforcer.enforceOnParent(EntityType.DATASET, namespace,
```

Comment:
dataset is a specific entity related to the data/datasets APIs, it is not related to writing to the database. This seems like it should be more like a list permission on the repo or namespace?

Reply:
yes correct, we need to add it in a separate PR